### ISSUE TITLE
Bug: Always syncing all related content with subscribed peers

### ISSUE THREAD

**Issue by juligasa** [seed-hypermedia/seed#14]
## Problem

Whenever there is an update on a document we are subscribed to, we download all the blobs, not only the ones we don't have.

## Steps to Reproduce

Even though we populate the list of haves before reconciling, the initiator seems to download all the data always, not only the new stuff

## Expected Behavior

Download only the new stuff

**Comment by juligasa** [comment]
At the very least we should first check if we already have the CID in our blockstore before calling GetBlock on the bitswap session.
If we have it, no need to fetch it again. 
Julio — Today at 4:18 PM
But I thougt that was the whole purpose of the reconciliation algo. That we ended up with a list of wants that we did not have 
Alex Burdiyan — Today at 4:21 PM
Yes. But we still should do this check, because we can get this blob from some other reconciliation, e.g. in parallel we could be syncing a different branch of this document, or the same document can be a reference material or a backlink of many different documents. So the same blob can be a want from multiple concurrent reconciliations. 
What's happening right now is probably (most definitely) a bug somewhere. Either the ordering of the blobs is not consistent on both peers, or something is off between them which makes their ranges not match.
Their ranges will never match, even if the order is the same

If the remote has the set of {a b c d x y z} but we have {a a1 a2 b b1 b2 c d e f g h i j k l m o p q r s t x y z} whatever fingerprint range the remote will send us will never match our local fingerprint, because we will never have the same dataset.
All of this is not a problem if we skip the blobs we already have.
